base-server/index.js

```diff
@@ -355,6 +372,45 @@ export class BaseServer {
     this.listenNotes = {}
     bindBackendProxy(this)
 
+    let end = (actionId, queue, queueKey, ...args) => {
+      this.actionToQueue.delete(actionId)
+      if (queue.length() === 0) {
+        this.queues.delete(queueKey)
+      }
+      queue.next(...args)
+    }
+    let undoRemainingTasks = queue => {
+      let remainingTasks = queue.getQueue()
+      if (remainingTasks) {
+        for (let task of remainingTasks) {
+          this.undo(task.action, task.meta, 'error')
+          this.actionToQueue.delete(task.meta.id)
+        }
+      }
+      queue.killAndDrain()
+    }
+    this.on('error', (e, action, meta) => {
+      let queueKey = this.actionToQueue.get(meta?.id)
+      if (!queueKey) {
+        return
+      }
+      let queue = this.queues.get(queueKey)
+      undoRemainingTasks(queue)
+      end(meta.id, queue, queueKey, e)
+    })
+    this.on('processed', (action, meta) => {
+      let queueKey = this.actionToQueue.get(action?.id)
```

Comment:
You can’t check `action.id` until you check `action.type`. Diferent actions could have a very dfferent data in `action` with possible collisions.

So add `logux/processed` or something here. But I think we should check `meta.id` to mark tasks as processed on their own `processed` event, not on `processed` event of `logux/processed` action for that action.